integration-tests/pkg/mock_sensor/expect_conn.go

```diff
@@ -72,14 +72,67 @@ loop:
 			if conn.GetContainerId() != containerID {
 				continue loop
 			}
-
 			if len(s.Connections(containerID)) == n {
 				return s.Connections(containerID)
 			}
 		}
 	}
 }
 
+// checkIfConnectionsMatchExpected compares a list of expected and observed connection match exactly.
+func (s *MockSensor) checkIfConnectionsMatchExpected(t *testing.T, connections []types.NetworkInfo, expected []types.NetworkInfo) bool {
```

Comment:
I think this function is redundant now because the extra logic is already handled by `collectorAssert.ElementsMatchFunc`, so in `ExpectSameElementsConnections` below, we should be able to just do:

```
collectorAssert.AssertElementsMatchFunc(t, expected, connections, func(a, b types.NetworkInfo) bool {
    return a.Equals(b)
})
```

If we also make the receiver for types.NetworkInfo.Equal a value rather than a pointer, we can further simplify:

```
collectorAssert.AssertElementsMatchFunc(t, expected, connections, types.NetworkInfo.Equal)
```

we'd get a copy I think, but that's almost certainly fine for the scale we have in the tests

Reply:
In fact, we already copy because the function passed to `ElementsMatchFunc` takes the variables by value anyway, so changing types.NetworkInfo.Equal would really just make things a little simpler